src/features/dialog-manager/reducer.js

```diff
@@ -40,14 +42,40 @@ const initialState = {
     },
     characterCreation: false,
     character: {
-        characterName: '',
+        characterName: ' ',
         characterRace: 'Human',
         characterClass: 'Fighter',
     },
 };
 
 const dialogManagerReducer = (state = initialState, { type, payload }) => {
     const { abilities, abilities_minimum, character } = state;
+    // if in character creation mode
+    if (state.characterCreation && character.characterName !== ' ') {
+        resetAbilityScoreValues(abilities);
+        switch (character.characterRace) {
+            case 'Human':
+                abilities.strength += RACE_ABILITY_BONUS;
+                abilities.intelligence += RACE_ABILITY_BONUS;
+                abilities_minimum.min_strength = RACE_ABILITY_BONUS;
+                abilities_minimum.min_intelligence = RACE_ABILITY_BONUS;
+                break;
+            case 'Elf':
+                abilities.dexterity += RACE_ABILITY_BONUS;
+                abilities.charisma += RACE_ABILITY_BONUS;
+                abilities_minimum.min_dexterity = RACE_ABILITY_BONUS;
+                abilities_minimum.min_charisma = RACE_ABILITY_BONUS;
+                break;
+            case 'Dwarf':
+                abilities.constitution += RACE_ABILITY_BONUS;
+                abilities.wisdom += RACE_ABILITY_BONUS;
+                abilities_minimum.min_constitution = RACE_ABILITY_BONUS;
+                abilities_minimum.min_wisdom = RACE_ABILITY_BONUS;
+                break;
+            default:
+        }
+    }
+
```

Comment:
Could this be made into a function, or refactored in some way? It seems very out of place where it is right now. That being said, I'm not entirely sure what change would be required to improve it and make the code cleaner whilst preserving the functionality.

Reply:
What part of it is out of place for you? The switch case or the repeated 4 lines under a case? Depending on where the issue is, I can refactor into a function probably from there.